src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java

```diff
@@ -77,6 +77,18 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
       LspLog.println("Workspace root was null");
     }
 
+    if (params.getWorkspaceFolders() == null) {
+      try {
+        File temp = Files.createTempDirectory("tmp-smithy-language-server-workspace").toFile();
+        System.out.println("Created temporary workspace root: " + temp);
+        temp.deleteOnExit();
+        WorkspaceFolder workspaceFolder = new WorkspaceFolder(temp.toURI().toString());
+        params.setWorkspaceFolders(ImmutableList.of(workspaceFolder));
```

Comment:
An editor could add/remove workspace folders at any time and I'm not sure if lsp4j will defensively copy this list. If it doesn't and tries to modify the same list that'll be a sure bug (exception thrown) later. If it's too hard to verify should be happening here I'd be more comfortable with a mutable list instead.

Reply:
Switched to `ListUtils.of` in https://github.com/awslabs/smithy-language-server/pull/54/commits/033ddbafabccfaea4a295f273e9d0e89811b9c77.